_i18n/en/contribute/develop/app.md

```diff
@@ -6,11 +6,93 @@ There's a few areas where we could well use your support. When first getting sta
+Before submitting a PR, always *announce your interest first*. Sometimes we haven't reached an agreement on a user experience and interface yet. Or it can happen that someone volunteered to start working on something, but we forgot to 'assign' the GitHub issue. Chipping in on our [forum](https://forum.antennapod.org) is the best way to avoid losing hours on code that won't be accepted.
 
-**During** Have a look at our wiki; it contains [handy info for contributors](https://github.com/AntennaPod/AntennaPod/wiki). You'll find there, for example, information about the project's [code style](https://github.com/AntennaPod/AntennaPod/wiki/Code-style) and can learn [how to make sure you get accredited](https://github.com/AntennaPod/AntennaPod/wiki/Getting-accredited-on-the-Contributors-page) for your work in the app. Last but not least: build instructions and info about debug logs.
+To keep things tidy, we prefer to have broader discussions on features on our forum. Once a decision has been reached, a concrete and boiled down description of the desired solution will be added to the relevant issue. More technical discussions on how to implement that can then take place in pull requests.
 
-**After** Talk about your work, and claim that fame! It might inspire other potential contributors, and helps promote the app generally. We're happy to retweet your tweets!
+## Code structure & code style
+AntennaPod is structured in several modules to ensure that similar code is close together and to avoid cyclic dependencies, which makes sure that the code can easily be tested. As an initial overview, the figure below gives the structure of all AntennaPod modules.
+
+{% include image.html
+   alt="A graph with 31 ovals and a recognisable set of arrows between them."
+   loc="/assets/images/blog/2024"
+   file="codeStructureAfter.png"
+%}
+
+For the Java code, AntennaPod uses a style similar to [Google's Java style](https://google.github.io/styleguide/javaguide.html). Checkstyle is run automatically when creating a pull request. The Gradle `checkstyle` task only checks more relaxed rules ([config/checkstyle/checkstyle.xml](https://github.com/AntennaPod/AntennaPod/blob/develop/config/checkstyle/checkstyle.xml)) for the whole code base. On CI, the more strict rules ([config/checkstyle/checkstyle-new-code.xml](https://github.com/AntennaPod/AntennaPod/blob/develop/config/checkstyle/checkstyle-new-code.xml)) are used for new code. In Android Studio, you can set the checkstyle version to 8.18 and select the [config/checkstyle/checkstyle-new-code.xml](https://github.com/AntennaPod/AntennaPod/blob/develop/config/checkstyle/checkstyle-new-code.xml) config file. Please only reformat lines that you actually modify, not the whole file. This makes it easier to read the version history.
```

Comment:
We no longer have the `checkstyle-new-code.xml` file since quite a while. Actually, only the first two sentences are still relevant